lib/hexo/index.js

```diff
@@ -135,9 +135,19 @@ Hexo.prototype._bindLocals = function() {
     return db.model('Page').find(query);
   });
 
-  locals.set('categories', () => db.model('Category'));
+  locals.set('categories', () => {
+    // Ignore categories with zero posts
+    return db.model('Category').filter(category => {
+      return category.length !== 0;
+    });
+  });
 
-  locals.set('tags', () => db.model('Tag'));
+  locals.set('tags', () => {
+    // Ignore tags with zero posts
+    return db.model('Tag').filter(tag => {
+      return tag.length !== 0;
+    });
```

Comment:
```
tag => {
      return tag.length !== 0;
    }
```
Or
```
tag => tag.length
```
🤔

Reply:
Does `tag => tag.length` works?

Reply:
Yes, both `tag => tag.length` and `tag => { return tag.length !== 0; }` are working properly

Reply:
Prefer `tag => tag.length` then.

Reply:
Done